### PR TITLE
Fix: Prevent consecutive operator input in CalculatorModal

### PR DESCRIPTION
**Pull request by akashs056**
## Pull request (PR) checklist
Please check if your pull request fulfills the following requirements:

- [x] I've read the [Contribution Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/CONTRIBUTING.md) and my PR doesn't break the rules.
- [x] I've read and understand the [Developer Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/docs/Guidelines.md).
- [x] I confirm that I've run the code locally and everything works as expected.
- [x] My PR includes only the necessary changes to fix the issue (i.e., no unnecessary files or lines of code are changed).
- [x] 🎬 I've attached a **screen recording** of using the new code to the next paragraph (if applicable).

## What's changed?
Resolved an issue in the CalculatorModal where consecutive operators could be added, resulting in invalid expressions.

- Updated the handleOperator function to ensure only one operator is added at a time.
- Replaced the last operator if a new operator is pressed.
- Ensured the input behaves like a standard calculator.

This change improves the user experience by preventing invalid expressions and aligning the calculator's behavior with user expectations.

Before
https://github.com/user-attachments/assets/c4cb087d-e360-44b1-90fd-38f3436e5e04

After:
https://github.com/user-attachments/assets/085816dc-9ad7-459a-b48e-c35e3bd649ad

### CODE REVIEW

File: temp/legacy-code/src/main/java/com/ivy/legacy/legacy/ui/theme/modal/edit/CalculatorModal.kt [hunk 1]
```diff
@@ -178,6 +178,17 @@ fun BoxWithConstraintsScope.CalculatorModal(
...
+private fun handleOperator(expression: String, operator: String): String {
+    return if (expression.isNotEmpty() && expression.last().isOperator()) {
+        expression.dropLast(1) + operator
+    } else {
+        expression + operator
+    }
+}
+
+private fun Char.isOperator(): Boolean {
+    return this == '+' || this == '−' || this == '×' || this == '÷'
```
Comment (ILIYANGERMANOV) [inline temp/legacy-code/src/main/java/com/ivy/legacy/legacy/ui/theme/modal/edit/CalculatorModal.kt:190]: Use 
```
fun Char.isOperator(): Boolean = when(this) {
  '+',  '-' -> true 
  else -> false
}
```

File: temp/legacy-code/src/main/java/com/ivy/legacy/legacy/ui/theme/modal/edit/CalculatorModal.kt [hunk 2]
```diff
@@ -178,6 +178,17 @@ fun BoxWithConstraintsScope.CalculatorModal(
         Spacer(Modifier.height(24.dp))
     }
 }
+private fun handleOperator(expression: String, operator: String): String {
```
Comment (ILIYANGERMANOV) [inline temp/legacy-code/src/main/java/com/ivy/legacy/legacy/ui/theme/modal/edit/CalculatorModal.kt:181]: Add new line above

**Review by ILIYANGERMANOV — COMMENTED** [review]
LGTM! Left some minor comments to resolve. Does it work for entering "-10"? Lots of people use it for adding negative balances

**Comment by akashs056** [comment]
> Does it work for entering "-10"

I dont get it .......

File: temp/legacy-code/src/main/java/com/ivy/legacy/legacy/ui/theme/modal/edit/CalculatorModal.kt (hunk 1 above)
Comment (akashs056) [reply temp/legacy-code/src/main/java/com/ivy/legacy/legacy/ui/theme/modal/edit/CalculatorModal.kt:190]: > ```kotlin
> fun Char.isOperator(): Boolean = when(this) {
>   '+', '-' -> true 
>   else -> false
> }
> ```

if we use this then the ' x ' and ' ÷ ' will get added consecutively

**Comment by ILIYANGERMANOV** [comment]
> > Does it work for entering "-10"
> 
> I dont get it .......

Can you input the following expression when adjusting balance?

```
-10.32
```

**Comment by akashs056** [comment]
> > > Does it work for entering "-10"
> > 
> > 
> > I dont get it .......
> 
> Can you input the following expression when adjusting balance?
> 
> ```
> -10.32
> ```


https://github.com/user-attachments/assets/0ba36d1a-0f07-442b-a171-814809013816

If you mean this then yes

File: temp/legacy-code/src/main/java/com/ivy/legacy/legacy/ui/theme/modal/edit/CalculatorModal.kt (hunk 1 above)
Comment (akashs056) [reply temp/legacy-code/src/main/java/com/ivy/legacy/legacy/ui/theme/modal/edit/CalculatorModal.kt:190]: > yeah for all operators was lazy to write, just use `when` so it's more idiomatic and simple

      fun Char.isOperator(): Boolean = when(this) {
    '+', '-','×', '÷' -> true
    else -> false
      }

is this ok ?

Comment (ILIYANGERMANOV) [reply temp/legacy-code/src/main/java/com/ivy/legacy/legacy/ui/theme/modal/edit/CalculatorModal.kt:190]: yup

**Comment by akashs056** [comment]
It's ready to merge

**Comment by ILIYANGERMANOV** [comment]
@akashs056 you missed to add which is closed by this PR and now I have to manually go and close it. Not a big deal but so you know for future PRs

**Pull request merged**